enhancements/template.md

```diff
@@ -193,6 +193,22 @@ enhancement:
 - What changes (in invocations, configurations, API use, etc.) is an existing
   cluster required to make on upgrade in order to make use of the enhancement?
 
+Upgrade expectations:
```

Comment:
Do we just want to set exceptions here, or do we want specific questions?

admittedly, people already mostly ignore the entire template after the first or second question....

Reply:
Not sure I understand your suggestion

Reply:
oh, I see now - this guidance could be expressed in terms of questions about the upgrade aspects of this enhancement.

e.g.

- How will you ensure it is possible to ugprade from the previous minor release to the minor release which introduces this enhancement?

Maybe ... but my attempts just now are pretty clunky. The question is at the top of this section, and the rest is background information to help people answer?

Reply:
I like the mix where it poses the question and then gives guidance on how to think about the answer.  I've seen a lot of enhancements where this section is either ignored, or only given trivial attention.  I'm hoping the additional guidance will help generate more thought and discussion!